### PR TITLE
MacOS Qt 5.15.2

### PR DESCRIPTION
**Pull request by ArcadeFever**
Reverted two changes from last PR

### CODE REVIEW

File: tools/qt-builder/README.md [hunk 1]
```diff
@@ -288,44 +285,40 @@ Zip up this directory and upload it to Backtrace or other crash log handlng tool
...
-`cd ../qt5-install`  
+`cd` to the `qt5-install directory`  
 `find . -name \*.prl -exec sed -i -e '/^QMAKE_PRL_BUILD_DIR/d' {} \;`  
 `cd ..`  
-1.   Copy *qt.conf* to *qt5-install\bin*
+1.   Note: you may have additional files in qt5-install/lib and qt5-install/lib/pkg/pkgconfig that have your local build absolute path included.  Optionally you can fix these as well, but it will not effect the build if left alone.
+
+Add a *qt.conf* file.  
+1. Copy the file *qt5-build\qtbase\bin\qt.conf* to *qt5-install\bin*
+1. Edit the *qt.conf* file: replace all absolute URLs with relative URLs (beginning with .. or .)
```
Comment (JulianGro) [inline tools/qt-builder/README.md:314]: You are using `1.` four times in the same section

File: tools/qt-builder/README.md [hunk 2]
```diff
@@ -104,16 +104,13 @@ Verify again
...
+### Mac  
 1.  git >= 1.6  
 Check if needed `git --version`  
 Install from https://git-scm.com/download/mac  
 Verify again  
-1.  pkg-config  
-brew fontconfig dbus-glib stall pkg-config  
-1.  dbus-1  
-brew install dbus-glib  
+1.  install pkg-config, dbug-glib, and fontconfig
```
Comment (JulianGro) [inline tools/qt-builder/README.md:112]: You are using `1.` twice here.

**Review by JulianGro — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-8782504.tar_bebb32ff.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-8782504_bebb32ff.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-8782504.tar_bebb32ff.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-8782504_bebb32ff.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-8782504_bebb32ff.exe

File: tools/qt-builder/README.md (hunk 1 above)
Comment (ArcadeFever) [reply tools/qt-builder/README.md:314]: I followed same markup notation used in windows section, which renders correctly although in source looks to be 1. multiple times.  Please refer to original document source before my change.

File: tools/qt-builder/README.md (hunk 2 above)
Comment (ArcadeFever) [reply tools/qt-builder/README.md:112]: I followed same markup notation used in windows section, which renders correctly although in source looks to be 1. multiple times.  Please refer to original document source before my change.

**Review by JulianGro — APPROVED** [review]
I cannot test it, but everything seems perfectly appropriate.

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by ctrlaltdavid** [comment]
I checked the PR's OSX build logs and it's downloading and using the new Qt 5.15.2 package OK.
I downloaded and tested the Windows PR build - Interface and server still work.

The OSX PR build should be downloaded and tested.

**Comment by ArcadeFever** [comment]
I tried PR968 MacOS-latest build, and at first it crashed on me, but after reboot its now working fine.  I'd like more people to try this PR build just to be sure everything is working.

**Review by digisomni — APPROVED** [review]
On a Mac Mini (late 2014) with Catalina 10.15.7 I was able to visit the serverless tutorial and the Hub without crashing. It appeared to be alright for me.

**Comment by ctrlaltdavid** [comment]
We need a test on a Mac that doesn't have a build environment installed.

**Comment by digisomni** [comment]
Eric cannot get this to run on his Mac. Snapshots:

image
image

**Comment by digisomni** [comment]
Update: after instructing him to try https://support.apple.com/en-us/HT202043 and running in dedicated only mode, it worked. Will report if there are further issues.

So far this PR looks mergeable.

**Comment by ctrlaltdavid** [comment]
Can Interface detect the condition that causes the crash and , perhaps, display a dialog box saying "not running because ..." then exit? (E.g., similar to how it does if OpenGL version is too old.)

**Comment by digisomni** [comment]
The previous release was running fine. I suppose there are enough differences with the upgrade that cause some weird detection to happen. It is entirely possible to be an issue with our threading or detection logic.

Ideally, though, crashes would exit with error codes instead of dying straight to desktop without a how or why.

**Comment by JulianGro** [comment]
Vircadia should run fine on those integrated graphics so I really wouldn't say that needing to manually switch to dedicated graphics is fine.

**Comment by digisomni** [comment]
It should, and it does, the problem is probably with the switching/threading/something Qt related. We're going to try and narrow it down. It appears to be WebEngine related however to test this as a developer live I need a Macbook Pro. lol

**Comment by JulianGro** [comment]
Cannot they just run LLDB?
Apparently it is pretty compatible with GDB, so you would just do something like 
`lldb interface.app`
`run`
and when it crashed `bt` and post the output

**Comment by ArcadeFever** [comment]
Oh, thanks for the pointer to LLDB.  I've been having a heck of a time trying to get GDB to run properly (keep getting Mach task error) and all the guides I follow on setting up certificates in the key ring don't seem to help.  Anyway, I just tried LLDB and it worked off the bat, so I'm ready to debug this.....but I can't get it to crash, even with the discrete card switch turned on. ¯\_(ツ)_/¯

**Comment by JulianGro** [comment]
Is it maybe a specific graphics setting? Have you tried both low and high setting?
Is it maybe a specific domain? Habe you tried the Hub and the serverless tutorial?

**Comment by ArcadeFever** [comment]
Adding my configure command to document what features were included in the MacOS QT 5.15.2 library.
[MacOS-QT-5.15.2-Configure.txt](https://github.com/vircadia/vircadia/files/5899334/MacOS-QT-5.15.2-Configure.txt)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-51b99d8.tar_97fc83ba.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-51b99d8.tar_97fc83ba.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-51b99d8_97fc83ba.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-51b99d8_97fc83ba.dmg

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Quest-Alpha-PR968-51b99d8_97fc83ba.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-51b99d8_97fc83ba.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR968/Vircadia-Alpha-PR968-51b99d8_97fc83ba.exe

**Comment by ctrlaltdavid** [comment]
Crash on start-up:
- Happens for some people all the time - Eric (unless graphics cards are enabled/disbled)
- Happens for some people every second time - Steve
- Doesn't happen for some people - Dave K

Am merging thing PR into the `qt-update` branch, which I will subsequently update to latest `master`, and we can address OSX crash-on-startup from there.

**Pull request merged**